internal/controllers/core/tiltfile/reconciler.go

```diff
@@ -309,8 +309,10 @@ func (r *Reconciler) run(ctx context.Context, nn types.NamespacedName, tf *v1alp
 		tlr.Error = fmt.Errorf("No resources found. Check out https://docs.tilt.dev/tutorial.html to get started!")
 	}
 
-	if tlr.Orchestrator() != model.OrchestratorUnknown {
-		r.dockerClient.SetOrchestrator(tlr.Orchestrator())
+	if tlr.HasOrchestrator(model.OrchestratorK8s) {
```

Comment:
have you seen this code?
https://github.com/tilt-dev/tilt/blob/master/internal/cli/doctor.go#L54

what might be good is to have a little bit of code to map orchestrators to docker clients.

if both orchestrators use the same docker client, then it's ok to use them both at the same time.

does that make sense?

Reply:
Nope, haven't seen it! Thanks for the pointer.

My changes to the "switch" docker client were an attempt to deal with the multiple docker clients without fully understanding the consequences there. It also seems like with multiple Docker clients the image builder needs to be aware of where the image is going to be deployed in order to choose the correct client, and I haven't chased that down yet other than forcing anything going through the DockerComposeBuildAndDeployer to use the local docker client.

This could also end up being another learning PR for me, as a way of exploring obstacles to correctly using multiple docker clients.

Reply:
fwiw - i think we should try to move forward with this PR. 

allowing both DC+K8s when there's a single docker client is a good goal!

have you tested manually if we push to the right registry when you have both DC+K8s in-play? I _think_ it should work (bc DC image builds are a different codepath)

Reply:
Yes! The example above works, the image is pushed to the Kind registry.